### DISCUSSION TITLE
How to return values from update statement using query/executeRaw

### DISCUSSION THREAD

**Discussion by joshkostal** [prisma/prisma#23332] [Q&A]
### Question

I am trying to return data from an update statement that has to be done in `queryRaw` or `executeRaw`. I run the query and nothing seems to happen. No errors occur, no data is updated, and nothing is returned. Here's the query:

```
with cte as (
        select
          id
        from table_a a
        join table_b b
          on a.id = b.a_id
          and b.user_id is null
        where b.token = '_'
        limit 1
      )
      update table_a a
      set user_id = _,
        name = '_'
      from cte
      where a.id = cte.id
      returning
        a.x_id,
        a.id;
```

The query runs fine in Postico, so it must be something with Prisma. Any tips on what I'm doing wrong?

### Expected behavior (optional)

Should return fields `x_id` and `id`.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @joshkostal 👋 

Thank you for raising this question.

Can you please share the relevant parts of your schema file so I can try reproducing this?

**Reply by joshkostal** [reply]
Here's the query:

```
const [{
      fantasy_league_id,
      id,
    } = {}] = await db.$queryRaw<{
      fantasy_league_id: number;
      id: number;
    }[]>`
      with team_to_update as (
        select
          ft.id
        from "FantasyTeam" ft
        join "FantasyLeague" fl
          on fl.id = ft.fantasy_league_id
          and ft.user_id is null
        where fl.token = '${fantasyLeagueToken}'
        limit 1
      )
      update "FantasyTeam" ft
      set user_id = ${userId},
        name = '${fantasyTeamName}'
      from team_to_update
      where ft.id = team_to_update.id
      returning
        ft.fantasy_league_id,
        ft.id;
    `;
```

Here's the schema for `FantasyTeam`:

```
CREATE TABLE "FantasyTeam" (
    id integer DEFAULT nextval('"FantasyTeam_id_seq"'::regclass) PRIMARY KEY,
    name text NOT NULL,
    fantasy_league_id integer NOT NULL REFERENCES "FantasyLeague"(id) ON DELETE RESTRICT ON UPDATE CASCADE,
    user_id integer REFERENCES "User"(id) ON DELETE SET NULL ON UPDATE CASCADE
);
```

and the schema for `FantasyLeague`:

```
CREATE TABLE "FantasyLeague" (
    id integer DEFAULT nextval('"FantasyLeague_id_seq"'::regclass) PRIMARY KEY,
    name text NOT NULL,
    commissioner_id integer NOT NULL REFERENCES "User"(id) ON DELETE RESTRICT ON UPDATE CASCADE,
    team_count integer NOT NULL,
    token character(15) NOT NULL,
    league_id integer NOT NULL REFERENCES "League"(id) ON DELETE RESTRICT ON UPDATE CASCADE
);
```

**Reply by ludralph** [reply]
Hi @joshkostal 

Can you please share the `User` table as well as I'm running into an error `ERROR:  relation "User" does not exist`. This will help me in reproducing this correctly.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
